sdk/state/ingest.go

```diff
@@ -108,3 +115,37 @@ func (c *Channel) ingestTxToUpdateUnauthorizedCloseAgreement(tx *txnbuild.Transa
 
 	return nil
 }
+
+func (c *Channel) ingestTxToUpdateBalances(tx *txnbuild.Transaction) error {
+	localEscrowAddress := c.localEscrowAccount.Address.Address()
+	remoteEscrowAddress := c.remoteEscrowAccount.Address.Address()
+
+	for _, op := range tx.Operations() {
+		if paymentOp, ok := op.(*txnbuild.Payment); ok {
```

Comment:
There are other operations that an credit an escrow account. Claim claimable balance, account merge, path payment strict receive, path payment strict send, maybe others. This is an example of why reading the current balance out of the meta XDR may be more resilient to all the variations.

Reply:
I'll take a look at the meta XDR way. I thought about this and I was thinking the only way an escrow account balance might change is from a `Withdrawal` or `Deposit` defined in the SEP, which is a payment operation.

but if the meta XDR is a safer all encompassing way then makes sense to me

Reply:
That would be true for withdrawal, but deposit can happen using any operation. The method of deposit isn't defined in the SEP and can take any form since the channel can't limit how the deposit occurs.

Reply:
ah I took `"a standard payment operation"` literally as only a payment op, you're right it can be other types

Reply:
playing around in lab it looks like the ResultMetaXdr _could_ work. Here are the results for a couple deposit types, where account `GD4...3II` increases in balance:
[Payment Op](https://laboratory.stellar.org/#xdr-viewer?input=AAAAAgAAAAIAAAADAA%2F9%2FQAAAAAAAAAAWPnYf%2B6kQN3t44vgesQdWh4JOOPj7aer852I7RJhtzAAAAAXJLOWLwANrPwAAAAFAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAABAA%2F9%2FQAAAAAAAAAAWPnYf%2B6kQN3t44vgesQdWh4JOOPj7aer852I7RJhtzAAAAAXJLOWLwANrPwAAAAGAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAABAAAABAAAAAMAD%2F18AAAAAAAAAAD49aUpVx7fhJPK6wDdlPJgkA1HkAi85qUL1tii8YSZzQAAABddU3scAA%2F8sgAAAAEAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAEAD%2F39AAAAAAAAAAD49aUpVx7fhJPK6wDdlPJgkA1HkAi85qUL1tii8YSZzQAAABdjSVwcAA%2F8sgAAAAEAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAMAD%2F39AAAAAAAAAABY%2Bdh%2F7qRA3e3ji%2BB6xB1aHgk44%2BPtp6vznYjtEmG3MAAAABcks5YvAA2s%2FAAAAAYAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAEAD%2F39AAAAAAAAAABY%2Bdh%2F7qRA3e3ji%2BB6xB1aHgk44%2BPtp6vznYjtEmG3MAAAABcevbUvAA2s%2FAAAAAYAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAA%3D&type=TransactionMeta&network=test)
[Path Payment Send](https://laboratory.stellar.org/#xdr-viewer?input=AAAAAgAAAAIAAAADAA%2F8wAAAAAAAAAAAWPnYf%2B6kQN3t44vgesQdWh4JOOPj7aer852I7RJhtzAAAAAXSHbm1AANrPwAAAACAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAABAA%2F8wAAAAAAAAAAAWPnYf%2B6kQN3t44vgesQdWh4JOOPj7aer852I7RJhtzAAAAAXSHbm1AANrPwAAAADAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAABAAAABAAAAAMAD%2FzAAAAAAAAAAABY%2Bdh%2F7qRA3e3ji%2BB6xB1aHgk44%2BPtp6vznYjtEmG3MAAAABdIdubUAA2s%2FAAAAAMAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAEAD%2FzAAAAAAAAAAABY%2Bdh%2F7qRA3e3ji%2BB6xB1aHgk44%2BPtp6vznYjtEmG3MAAAABc%2FhhVUAA2s%2FAAAAAMAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAMAD%2FyyAAAAAAAAAAD49aUpVx7fhJPK6wDdlPJgkA1HkAi85qUL1tii8YSZzQAAABdIdugAAA%2F8sgAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAEAD%2FzAAAAAAAAAAAD49aUpVx7fhJPK6wDdlPJgkA1HkAi85qUL1tii8YSZzQAAABdRZ7mAAA%2F8sgAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAA%3D&type=TransactionMeta&network=test)
[Claim Claimable Balance](https://laboratory.stellar.org/#xdr-viewer?input=AAAAAgAAAAIAAAADAA%2F9fAAAAAAAAAAA%2BPWlKVce34STyusA3ZTyYJANR5AIvOalC9bYovGEmc0AAAAXUWe5HAAP%2FLIAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAABAA%2F9fAAAAAAAAAAA%2BPWlKVce34STyusA3ZTyYJANR5AIvOalC9bYovGEmc0AAAAXUWe5HAAP%2FLIAAAABAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAABAAAABgAAAAMAD%2F1OAAAAAAAAAABY%2Bdh%2F7qRA3e3ji%2BB6xB1aHgk44%2BPtp6vznYjtEmG3MAAAABcks5aTAA2s%2FAAAAAUAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAEAD%2F18AAAAAAAAAABY%2Bdh%2F7qRA3e3ji%2BB6xB1aHgk44%2BPtp6vznYjtEmG3MAAAABcks5aTAA2s%2FAAAAAUAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAMAD%2F0SAAAABAAAAACOCqli4ggwTMj2RMhLLA9g5b6IK6dL2avxl7syBWvLfwAAAAEAAAAAAAAAAPj1pSlXHt%2BEk8rrAN2U8mCQDUeQCLzmpQvW2KLxhJnNAAAAAAAAAAAAAAAAC%2BvCAAAAAAAAAAABAAAAAQAAAABY%2Bdh%2F7qRA3e3ji%2BB6xB1aHgk44%2BPtp6vznYjtEmG3MAAAAAAAAAACAAAABAAAAACOCqli4ggwTMj2RMhLLA9g5b6IK6dL2avxl7syBWvLfwAAAAMAD%2F18AAAAAAAAAAD49aUpVx7fhJPK6wDdlPJgkA1HkAi85qUL1tii8YSZzQAAABdRZ7kcAA%2F8sgAAAAEAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAEAD%2F18AAAAAAAAAAD49aUpVx7fhJPK6wDdlPJgkA1HkAi85qUL1tii8YSZzQAAABddU3scAA%2F8sgAAAAEAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAA%3D&type=TransactionMeta&network=test)

the `ledger_entry_updated` shows the account's new balance. I'm not 100% on how this field is generated/how safe it is to rely on. I can ask the Horizon team.

Another thought I had was maybe the [ingest library](https://github.com/stellar/go/tree/master/ingest) could be a good fit here?

Reply:
Horizon includes the meta XDR for every transaction in it's JSON response. I don't think the state package needs a dependency on the ingest package or on the horizon response, rather we can just pass in the xdr object. Where the xdr comes from can be up to the agent, and in the agent I'll likely use the horizon JSON response.

Reply:
For example, the `result_meta_xdr` field: https://horizon.stellar.org/transactions/4185451ea7460167d9aca63bcf19710b5c683bb65cd9d4ee508ef02eeee74e89

Reply:
ok cool, sgtm for using the `result_meta_xdr`!